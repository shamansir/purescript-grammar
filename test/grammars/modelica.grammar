main :- model.
model :- [modelKW, ws, ident, params, ws, vars, ws, initKW, ws, init, ws, equationKW, ws, equations, ws, endKW].
params :- ["(", repSep(ident, commaSpace), ")"].

vars :- repSep(varDecl, ws).
init :- repSep(initEq, ws).
varDecl :- [var, ";"].
initEq :- [var, ws, "=", ws, expr, ";"].
expr :- (intLit | derivative | var).
var :- ident.
derivative :- [derKW, "(", ident, ")"].
equations :- repSep(equation, ws).
equation :- [expr, ws, "=", ws, expr, ";"].

alpha :- ([a-z] | [A-Z] | "_").
num :- [0-9].
intLit :- [[0-9], repSep([0-9], "")].
alphaNum :- (alpha | num).
ident :- [alpha, repSep((alphaNum | "."), "")].
ws :- repSep((" "|"\n"), "").
commaSpace :- [",", ws].
semicolonSpace :- [";", ws].
placeholder :- "???".
stringChar :- (^'"' | ['\\', '"']).

modelKW :- "model".
initKW :- "init".
equationKW :- "equation".
endKW :- "end".
derKW :- "der".
