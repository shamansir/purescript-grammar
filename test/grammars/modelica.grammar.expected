main :- model.
alpha :- ([a-z]|[A-Z]|"_").
alphaNum :- (alpha|num).
commaSpace :- [",",ws].
derKW :- "der".
derivative :- [derKW,"(",ident,")"].
endKW :- "end".
equation :- [expr,ws,"=",ws,expr,";"].
equationKW :- "equation".
equations :- repSep(equation,ws).
expr :- (intLit|derivative|var).
ident :- [alpha,repSep((alphaNum|"."),"")].
init :- repSep(initEq,ws).
initEq :- [var,ws,"=",ws,expr,";"].
initKW :- "init".
intLit :- [[0-9],repSep([0-9],"")].
model :- [modelKW,ws,ident,params,ws,vars,ws,initKW,ws,init,ws,equationKW,ws,equations,ws,endKW].
modelKW :- "model".
num :- [0-9].
params :- ["(",repSep(ident,commaSpace),")"].
placeholder :- "???".
semicolonSpace :- [";",ws].
stringChar :- (^'"'|['\\','"']).
var :- ident.
varDecl :- [var,";"].
vars :- repSep(varDecl,ws).
ws :- repSep((" "|"\n"),"").